sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -104,14 +103,6 @@ def test_doit_matrix():
     raises(ValueError, lambda: MatPow(X,-2).doit())
 
 
-def test_nonsquare():
-    A = MatrixSymbol('A', 2, 3)
-    B = ImmutableMatrix([[1, 2, 3], [4, 5, 6]])
-    for r in [-1, 0, 1, 2, S.Half, S.Pi, n]:
-        raises(NonSquareMatrixError, lambda: MatPow(A, r))
-        raises(NonSquareMatrixError, lambda: MatPow(B, r))
```

Comment:
This seems like is should raise to me since we do know that the matrices are not square.